iib/workers/tasks/build_merge_index_image.py

```diff
@@ -126,28 +128,26 @@ def _add_bundles_missing_in_source(
             '%s bundles have invalid version label and will be deprecated.', len(invalid_bundles)
         )
 
-    with set_registry_token(overwrite_target_index_token, source_from_index):
+    with set_registry_token(overwrite_target_index_token, target_index, append=True):
         is_source_fbc = is_image_fbc(source_from_index)
-    if is_source_fbc:
-        opm_registry_add_fbc(
-            base_dir=base_dir,
-            bundles=missing_bundle_paths,
-            binary_image=binary_image,
-            from_index=source_from_index,
-            container_tool='podman',
-            overwrite_from_index_token=overwrite_target_index_token,
-        )
-    else:
-        _opm_index_add(
-            base_dir=base_dir,
-            bundles=missing_bundle_paths,
-            binary_image=binary_image,
-            from_index=source_from_index,
-            overwrite_from_index_token=overwrite_target_index_token,
-            # Use podman until opm's default mechanism is more resilient:
-            #   https://bugzilla.redhat.com/show_bug.cgi?id=1937097
-            container_tool='podman',
-        )
+        if is_source_fbc:
+            opm_registry_add_fbc(
+                base_dir=base_dir,
+                bundles=missing_bundle_paths,
+                binary_image=binary_image,
+                from_index=source_from_index,
+                container_tool='podman',
+            )
+        else:
+            _opm_index_add(
+                base_dir=base_dir,
+                bundles=missing_bundle_paths,
+                binary_image=binary_image,
+                from_index=source_from_index,
+                # Use podman until opm's default mechanism is more resilient:
+                #   https://bugzilla.redhat.com/show_bug.cgi?id=1937097
+                container_tool='podman',
+            )
```

Comment:
Why do we want to call `opm_registry_add_fbc` and `_opm_index_app` without passing the `overwrite_from_index_token`?

Reply:
Correct token (docker config file) is set by `with set_registry_token(overwrite_target_index_token, target_index, append=True):` on line 131. 
Unfortunately we have these settings wrongly written inside of almost all function. We agreed on guild meeting that we will support "private registry" for source index as well (if it is the same as target index).That is why I removed the overwrite token before this call to make sure it is not wrongly set for incorrect repository (from_index is source index and we set token for target index in this API, therefore call inside of this function is can set token to different repository which breaks IIB). I would need more time to do proper refactoring of this.